projects/laji/src/app/shared/service/confirm-modal.component.ts

```diff
@@ -27,16 +27,16 @@ import { BsModalRef } from 'ngx-bootstrap/modal';
 })
 export class ConfirmModalComponent implements OnInit, AfterViewInit {
 
-  message: string;
+  message!: string;
   confirmLabel = 'OK';
   cancelLabel = 'cancel';
   prompt = false;
   promptValue = '';
   showCancel = true;
 
-  value: boolean | string | null;
+  value!: boolean | string | null;
```

Comment:
I'm wondering what does the exclamation mark means here? I thought that it means that we tell the compiler that it can't be null but then its type has null?

Reply:
Yeah that's what it does, it's the non-null assertion operator. It's indeed a bit confusing, but adding it makes the compiler shut up about this:

```
Error: projects/laji/src/app/shared/service/confirm-modal.component.ts:37:3 - error TS2564: Property 'value' has no initializer and is not definitely assigned in the constructor.

37   value: boolean | string | null;
 ```

So in this context, it kind of makes sense that we are telling the compiler that `value` will have a sane value (and we take responsibility of it by adding the non-nullish assertion) even though it's not initialized at declaration. It's similar to `any`: it makes the compiler to skip the type checkings and you take responsibility of the integrity of the typings. `any` and `!` could be considered antipatterns, but our goal is to just get the strict mode on, so it's justifiable here.

Reply:
Ok, makes sense, thanks for the explanation!